Refactor fs_connect function on client

rolysr opened the pull request:
These PR ensures that the `fs_connect` readability is improved by changing some variables names to more descriptive ones and also to set some magic numbers on the function as constants. The idea is to make it easier for developers to make some changes on the registration protocol function from client to master without expending much time on understanding `fs_connect`  implemention.

rolysr committed:
refactor: Renamed variables on `fs_connect`

This commit improves the readability of the `fs_connect` function
on the mastercomm.cc file for the client module, which helps on
future maintainability of the function.

Signed-off-by: Rolando Sánchez Ramos <rolysr@leil.io>

rolysr committed:
chore: Name magic numbers in `fs_connect`

This commit replaces hardcoded time values in the fs_connect
function with descriptive named constants. This improves
code readability and maintainability by making the meaning
of these values explicit.

Signed-off-by: Rolando Sánchez Ramos <rolysr@leil.io>

rolysr commented on src/mount/mastercomm.cc:
```diff
@@ -477,37 +489,50 @@ void fs_register_config() {
 	const char *sesflagnegstrtab[]={SESFLAG_NEG_STRINGS};
+	uint32_t registerSessionPacketLength =
+	    (gInitParams.meta) ? sizeof(REGISTER_NEWMETASESSION) : sizeof(REGISTER_NEWSESSION);
+	uint32_t passwordDigestLength = sizeof(passwordDigestLength);
```
Done.

rolysr commented on src/mount/mastercomm.cc:
```diff
@@ -538,123 +563,144 @@ int fs_connect(bool verbose) {
+		                   fuseRegisterBlobAclLength,
+		               kDefaultTcpCommTimeoutMSeconds) != cltomaFuseRegisterHeaderLength +
+		                                                      fuseRegisterBlobAclTotalPacketLength +
+		                                                      fuseRegisterBlobAclLength) {
```
Done.

rolysr commented on src/mount/mastercomm.cc:
```diff
@@ -762,49 +809,49 @@ int fs_connect(bool verbose) {
-				fprintf(stderr,"%s%s",j?",":"",sesflagnegstrtab[i]);
-				j=1;
+		bool clientHasSessionFlags=false;
+		for (int sesFlagsIndex=0 ; sesFlagsIndex<sizeof(sesflags) ; sesFlagsIndex++) {
```
Done.

rolysr commented on src/mount/mastercomm.cc:
```diff
@@ -477,37 +489,50 @@ void fs_register_config() {
+	    (gInitParams.meta) ? 0 : sizeof(subfolderPathLengthBytes);
+	uint32_t fuseRegisterPacketInfoLength =
+	    fuseRegisterSessionTypeSize + sizeof(mountPointPathLegthBytes) + saunafsVersionMajorSize +
+	    saunafsVersionMinorSize + saunafsVersionMinorSize + fuseRegisterSubfolderLength;
```
Done.

rolysr committed:
feat: Make password digest flexible on fs_connect

This changes the password digest handling to support variable-length
digests instead of a fixed 16-byte array. This change enables the
use of more secure and complex password hashes in the future,
allowing clients to send longer and more robust passwords to the
master server.

Signed-off-by: Rolando Sánchez Ramos <rolysr@leil.io>

dmga44 reviewed (commented):
These are my first suggestions:
- start the global constants with k.
- apply formatter to all the code of fs_connect
- split fs_connect function into some functions with meaningful names to better describe its doing. For now, the function looks way too long.

lgsilva3087 commented on src/mount/mastercomm.cc:
```diff
@@ -477,37 +489,49 @@ void fs_register_config() {
-	uint8_t digest[16];
-	const uint8_t *rptr;
+	uint32_t messageValueIterator;
+	uint8_t *messageToMaster,*registrationMessageBuffer;
```
Consider using std::vector (combined with .data()) for both, to avoid using malloc later.

rolysr replied:
Done.

lgsilva3087 commented on src/mount/mastercomm.cc:
```diff
@@ -477,37 +489,49 @@ void fs_register_config() {
 }
 
 int fs_connect(bool verbose) {
```
Apply the code style to the whole function.

rolysr replied:
Done.

lgsilva3087 commented on src/mount/mastercomm.cc:
```diff
@@ -477,37 +489,49 @@ void fs_register_config() {
+	uint32_t messageValueIterator;
+	uint8_t *messageToMaster,*registrationMessageBuffer;
+	md5ctx md5PassCtx;
+	std::vector<uint8_t> passwordDigest;
+	const uint8_t *messageFromMaster;
 	uint8_t havepassword;
-	uint32_t pleng,ileng;
+	uint32_t subfolderPathLengthBytes,mountPointPathLegthBytes;
 	uint8_t sesflags;
 	uint32_t rootuid,rootgid,mapalluid,mapallgid;
 	uint8_t mingoal,maxgoal;
 	uint32_t mintrashtime,maxtrashtime;
 	const char *sesflagposstrtab[]={SESFLAG_POS_STRINGS};
 	const char *sesflagnegstrtab[]={SESFLAG_NEG_STRINGS};
+	uint32_t passwordDigestLength = gInitParams.password_digest.size();
+	uint32_t fuseRegisterBlobAclPacketSizeValueLength = sizeof(fuseRegisterBlobAclLength);
+	uint32_t fuseRegisterBlobAclTotalPacketLength = fuseRegisterBlobAclLength + 1;
+	uint32_t fuseRegisterSessionTypeSize = 1U;
+	uint32_t fuseRegisterSubfolderLength =
+	    (gInitParams.meta) ? 0 : sizeof(subfolderPathLengthBytes);
+	uint32_t fuseRegisterPacketInfoLength =
+	    fuseRegisterSessionTypeSize + sizeof(mountPointPathLegthBytes) + saunafsVersionMajorSize +
+	    saunafsVersionMinorSize + saunafsVersionMicroSize + fuseRegisterSubfolderLength;
+	uint32_t masterResponseRegisterPacketLength = 32U;
```
Consider to separate these into logical blocks (separated by new line), to improve readability.

rolysr replied:
Done.

lgsilva3087 commented on src/mount/mastercomm.cc:
```diff
@@ -477,37 +489,49 @@ void fs_register_config() {
-		pleng=0;
-		regbuff = (uint8_t*) malloc(8+64+9+ileng+16);
+		subfolderPathLengthBytes=0;
+		registrationMessageBuffer = (uint8_t*) malloc(cltomaFuseRegisterHeaderLength+fuseRegisterBlobAclPacketSizeValueLength+fuseRegisterBlobAclLength+fuseRegisterPacketInfoLength+mountPointPathLegthBytes+passwordDigestLength);
```
Try to avoid the malloc if possible.

rolysr replied:
Done.

lgsilva3087 commented on src/mount/mastercomm.cc:
```diff
@@ -477,37 +489,49 @@ void fs_register_config() {
 	fd = tcpsocket();
 	if (fd<0) {
-		free(regbuff);
+		free(registrationMessageBuffer);
```
If you manage to convert it to std::vector, these `free` should not be needed.

rolysr replied:
Done.

lgsilva3087 commented on src/mount/mastercomm.cc:
```diff
@@ -664,73 +709,74 @@ int fs_connect(bool verbose) {
-	get32bit(&rptr, i);
-	if (!(i==1 || (gInitParams.meta && (i==5 || i==9 || i==19)) || (gInitParams.meta==0 && (i==13 || i==21 || i==25 || i==35)))) {
+	get32bit(&messageFromMaster, messageValueIterator);
+	if (!(messageValueIterator == 1 || (gInitParams.meta && (messageValueIterator == 5 || messageValueIterator == 9 || messageValueIterator == 19)) || (gInitParams.meta == 0 && (messageValueIterator == 13 || messageValueIterator == 21 || messageValueIterator == 25 || messageValueIterator == 35)))) {
```
Should be very nice if the magic numbers could be removed.

rolysr replied:
Done.

lgsilva3087 commented on src/mount/mastercomm.cc:
```diff
@@ -664,73 +709,74 @@ int fs_connect(bool verbose) {
 	}
-	if (i==9 || i==19 || i==25 || i==35) {
-		get32bit(&rptr, masterversion);
+	if (messageValueIterator==9 || messageValueIterator==19 || messageValueIterator==25 || messageValueIterator==35) {
```
Give some air (new line) between non related ifs to improve readability.

rolysr replied:
Done.

lgsilva3087 commented on src/mount/mastercomm.cc:
```diff
@@ -762,49 +808,49 @@ int fs_connect(bool verbose) {
+			struct group groupEntry,*groupInfo;
+			char userGroupBuffer[16384];
 
 			fprintf(stderr," ; root mapped to ");
```
Consider a string stream to concatenate the text and to print only at the end.
If possible, the final printing, with safs::log_* functions to have it in the syslog (Linux) or the client logs (Windows).

rolysr replied:
Done.

lgsilva3087 commented on src/mount/mastercomm.cc:
```diff
@@ -843,24 +889,24 @@ int fs_connect(bool verbose) {
 				fprintf(stderr," ; setgoal limited to (%u:%u)",mingoal,maxgoal);
 			}
-			if (mintrashtime>0 || maxtrashtime<UINT32_C(0xFFFFFFFF)) {
+			if (mintrashtime>0 || maxtrashtime<std::numeric_limits<uint32_t>::max()) {
```
These checks could be delegated to a new function. Maybe named clampTrashTime.

rolysr replied:
Done.

lgsilva3087 commented on src/mount/mastercomm.cc:
```diff
@@ -92,6 +92,18 @@ struct threc {
 #define NO_DATA_RECEIVED_FROM_MASTER NULL
 #define RECEIVE_TIMEOUT 10
 
+constexpr size_t kDefaultTcpCommTimeoutMSeconds = 1000;
+constexpr uint32_t fuseRegisterBlobAclLength = sizeof(FUSE_REGISTER_BLOB_ACL) - 1;
+constexpr uint32_t cltomaFuseRegisterHeaderLength = sizeof(CLTOMA_FUSE_REGISTER);
+constexpr size_t saunafsVersionMajorSize = 2U;
+constexpr size_t saunafsVersionMinorSize = 1U;
+constexpr size_t saunafsVersionMicroSize = 1U;
```
If some of these constants are only meant to be used from fs_connect, consider to declare them inside the function.

rolysr replied:
Done.

lgsilva3087 reviewed (commented):
Good direction. Please see my comments.

ralcolea commented on src/mount/mastercomm.cc:
```diff
@@ -477,37 +489,49 @@ void fs_register_config() {
+	uint8_t *messageToMaster,*registrationMessageBuffer;
+	md5ctx md5PassCtx;
+	std::vector<uint8_t> passwordDigest;
+	const uint8_t *messageFromMaster;
```
The variables declaration block is very large.
Please, consider declaring at the beginning of the function the variables that really need to be defined before being used.
Consider also split the whole block in different declaration blocks, grouped by the reponsibility of declared variables.

rolysr replied:
Done.

ralcolea commented on src/mount/mastercomm.cc:
```diff
@@ -477,37 +489,49 @@ void fs_register_config() {
+	const uint8_t *messageFromMaster;
 	uint8_t havepassword;
-	uint32_t pleng,ileng;
+	uint32_t subfolderPathLengthBytes,mountPointPathLegthBytes;
```
`mountPointPathLegthBytes` instead of `mountPointPathLengthBytes`.

rolysr replied:
Done.

ralcolea commented on src/mount/mastercomm.cc:
```diff
@@ -477,37 +489,49 @@ void fs_register_config() {
 
 	havepassword = !gInitParams.password_digest.empty();
-	ileng=gInitParams.mountpoint.size() + 1;
+	passwordDigestLength = gInitParams.password_digest.size();
```
`passwordDigest` is assigned twice with the same value (`gInitParams.password_digest.size()`), without being changed in between.
Consider to declare it and initialize it here instead of above.

rolysr replied:
Done.

ralcolea commented on src/mount/mastercomm.cc:
```diff
@@ -477,37 +489,49 @@ void fs_register_config() {
-		pleng=0;
-		regbuff = (uint8_t*) malloc(8+64+9+ileng+16);
+		subfolderPathLengthBytes=0;
+		registrationMessageBuffer = (uint8_t*) malloc(cltomaFuseRegisterHeaderLength+fuseRegisterBlobAclPacketSizeValueLength+fuseRegisterBlobAclLength+fuseRegisterPacketInfoLength+mountPointPathLegthBytes+passwordDigestLength);
```
Line initializing `registrationMessageBuffer` is extremely long.
Check the formatting in all the modified lines.

rolysr replied:
Done.

ralcolea commented on src/mount/mastercomm.cc:
```diff
@@ -538,123 +562,144 @@ int fs_connect(bool verbose) {
+		memcpy(messageToMaster,FUSE_REGISTER_BLOB_ACL,fuseRegisterBlobAclLength);
+		messageToMaster+=fuseRegisterBlobAclLength;
+		put8bit(&messageToMaster,REGISTER_GETRANDOM);
+		if (tcptowrite(fd, registrationMessageBuffer,
```
Consider caching in a dedicated variable the size of the packet to be sent over the network.
Consider separating all the code in different blocks to improve the readability.

rolysr replied:
Done.

ralcolea commented on src/mount/mastercomm.cc:
```diff
@@ -538,123 +562,144 @@ int fs_connect(bool verbose) {
 			return -1;
 		}
-		if (tcptoread(fd,regbuff,8,1000)!=8) {
+		if (tcptoread(fd, registrationMessageBuffer,
```
Same comment about using dedicated variables for the packet sizes.

rolysr replied:
Done.

ralcolea commented on src/mount/mastercomm.cc:
```diff
@@ -538,123 +562,144 @@ int fs_connect(bool verbose) {
-		get32bit(&rptr, i);
-		if (i!=32) {
+		get32bit(&messageFromMaster, messageValueIterator);
+		if (messageValueIterator!=masterResponseRegisterPacketLength) {
```
Consider to separate every `if - block` with empty lines to improve readability.

rolysr replied:
Done.

ralcolea commented on src/mount/mastercomm.cc:
```diff
@@ -538,123 +562,144 @@ int fs_connect(bool verbose) {
 	if (gInitParams.meta) {
 		if (havepassword) {
-			put32bit(&wptr,64+9+ileng+16);
+			put32bit(&messageToMaster,fuseRegisterBlobAclLength+fuseRegisterPacketInfoLength+mountPointPathLegthBytes+passwordDigestLength);
```
Line too long. Check the formatting and readability.
Check also in the `else` branch.

rolysr replied:
Done.

ralcolea commented on src/mount/mastercomm.cc:
```diff
@@ -538,123 +562,144 @@ int fs_connect(bool verbose) {
-	if (tcptowrite(fd,regbuff,8+64+(gInitParams.meta?9:13)+ileng+pleng+(havepassword?16:0),1000)!=(int32_t)(8+64+(gInitParams.meta?9:13)+ileng+pleng+(havepassword?16:0))) {
+		memcpy(messageToMaster+subfolderPathLengthBytes,passwordDigest.data(),passwordDigestLength);
+	}
+	if (tcptowrite(fd, registrationMessageBuffer,
```
Avoid computing long operations when passing function parameters.
Use dedicated variables to avoid duplicating the calculation of values and improve the readability.

rolysr replied:
Done.

ralcolea commented on src/mount/mastercomm.cc:
```diff
@@ -538,123 +562,144 @@ int fs_connect(bool verbose) {
 		return -1;
 	}
-	if (tcptoread(fd,regbuff,8,1000)!=8) {
+	if (tcptoread(fd, registrationMessageBuffer,
```
Same comment as before.

rolysr replied:
Done.

ralcolea commented on src/mount/mastercomm.cc:
```diff
@@ -664,73 +709,74 @@ int fs_connect(bool verbose) {
-			get32bit(&rptr, mapallgid);
+		get32bit(&messageFromMaster, rootuid);
+		get32bit(&messageFromMaster, rootgid);
+		if (messageValueIterator==21 || messageValueIterator==25 || messageValueIterator==35) {
```
Clarify if possible the meaning of the magic numbers.

rolysr replied:
Done.

ralcolea reviewed (commented):
Please, see some suggestions I shared that could be improved for the final version.

rolysr committed:
refactor: Add auxiliar functions on fs_connect

This commit places some of logic from fs_connect function on
separated axiliar functions in order to improve fs_connect
readability.

Signed-off-by: Rolando Sánchez Ramos <rolysr@leil.io>

dmga44 commented on src/mount/mastercomm.cc:
```diff
@@ -476,261 +488,457 @@ void fs_register_config() {
-			fd=-1;
-			free(regbuff);
-			return -1;
+	}
```
In this function, right before the returns, you can consider not to set the `socket` variable to -1 because you're returning the -1 anyway. You may also consider to return `socket` after being set to -1.

rolysr replied:
Done.

dmga44 commented on src/mount/mastercomm.cc:
```diff
@@ -476,261 +488,457 @@ void fs_register_config() {
+				fprintf(stderr, "can't bind socket to given ip (\"%s\")\n", srcstrip);
 			} else {
-				safs_pretty_syslog(LOG_WARNING,"error sending data to sfsmaster");
+				safs_pretty_syslog(LOG_WARNING, "can't bind socket to given ip (\"%s\")", srcstrip);
```
Consider to use `safs::log_warn()` instead of all the `safs_pretty_syslog(LOG_WARNING, "...");`

rolysr replied:
Done.

dmga44 commented on src/mount/mastercomm.cc:
```diff
@@ -476,261 +488,457 @@ void fs_register_config() {
+	return socket;
+}
+
+int fs_register_with_get_random(uint32_t &messageValueIterator, uint8_t *messageToMaster,
+                                const uint8_t *messageFromMaster,
+                                std::vector<std::uint8_t> &registrationMessageBuffer,
+                                std::vector<uint8_t> &passwordDigest, uint32_t passwordDigestLength,
+                                md5ctx &md5PassCtx, bool verbose) {
```
Consider not putting `messageToMaster` and `messageFromMaster` as parameters of the function as the first thing you do with them is to set them up equal to `registrationMessageBuffer.data()`. They could be local variables.

Similar suggestion goes to `messageValueIterator`, since you don't use its value in `fs_connect`, only here it could local variable. Also the `passwordDigestLength` parameter is the size of the `passwordDigest` vector, consider to make it local variable.

rolysr replied:
Done.

dmga44 commented on src/mount/mastercomm.cc:
```diff
@@ -476,261 +488,457 @@ void fs_register_config() {
+	return 0;
+}
+
+int fs_register_with_new_session(
+    uint32_t &messageValueIterator, uint8_t *messageToMaster, const uint8_t *messageFromMaster,
+    std::vector<std::uint8_t> &registrationMessageBuffer, std::vector<uint8_t> &passwordDigest,
+    uint32_t passwordDigestLength, uint32_t mountPointPathLength, uint32_t subfolderPathLength,
+    const uint32_t fuseRegisterPacketInfoLength, uint8_t havepassword, uint8_t &sesflags,
+    uint32_t &rootuid, uint32_t &rootgid, uint32_t &mapalluid, uint32_t &mapallgid,
+    uint8_t &mingoal, uint8_t &maxgoal, uint32_t &mintrashtime, uint32_t &maxtrashtime,
+    bool verbose) {
```
Consider analogous suggestion to the one for `fs_register_with_get_random`.

Plus, consider making `mountPointPathLength` and `subfolderPathLength` also local.

rolysr replied:
Done.

dmga44 commented on src/mount/mastercomm.cc:
```diff
@@ -476,261 +488,457 @@ void fs_register_config() {
+#ifndef _WIN32
+		struct passwd userEntry, *userInfo;
+		struct group groupEntry, *groupInfo;
+		char userGroupBuffer[16384];
```
Consider using `std::vector<char>` instead of an array of chars.

rolysr replied:
Done.

dmga44 reviewed (commented):
Consider to add small descriptions to the new functions and also `fs_connect`, like its purpose and return values.

That said, it is already looking much better :)

Please also take a look at my suggestions.

rolysr committed:
fix: Add more code review suggestions

Signed-off-by: Rolando Sánchez Ramos <rolysr@leil.io>

lgsilva3087 commented on src/mount/mastercomm.cc:
```diff
@@ -476,261 +488,465 @@ void fs_register_config() {
+			}
+			infoToPrint << ")";
+		}
+		fprintf(stderr, "%s\n", infoToPrint.str().c_str());
+		safs::log_info("{}", infoToPrint.str().c_str());
```
```fprintf(stderr,...``` prints to stderr, but
safs::log_info by default should print to stdout and to syslog.
Please verify that the output is not repeated in the terminal.

lgsilva3087 replied:
@uristdwarf could help here.

rolysr replied:
Done. Leaved as before and consider to modernizing most prints on another PR.

lgsilva3087 commented on src/mount/mastercomm.cc:
```diff
@@ -476,261 +488,465 @@ void fs_register_config() {
+                                          uint32_t mapalluid, uint32_t mapallgid) {
+	const uint32_t userGroupBufferDefaultSize = 16384;
+	const char *sesflagposstrtab[] = {SESFLAG_POS_STRINGS};
+	const char *sesflagnegstrtab[] = {SESFLAG_NEG_STRINGS};
```
Consider renaming these two variables.

rolysr replied:
Done.

lgsilva3087 commented on src/mount/mastercomm.cc:
```diff
@@ -476,261 +488,465 @@ void fs_register_config() {
+	uint32_t passwordDigestLength = passwordDigest.size();
+
+	// Constants for session parameters registration lengths
+	const uint32_t newMetaSessionDataLengthLegacy = 5;
+	const uint32_t newMetaSessionDataLengthWithPackageVersion = 9;
+	const uint32_t newMetaSessionDataLengthWithGoalsAndTrashtimes = 19;
+
+	const uint32_t newSessionDataLengthWithPackageVersion = 13;
+	const uint32_t newSessionDataLengthWithMapAllUidGid = 21;
+	const uint32_t newSessionDataLengthGoalsAndTrashtimes = 25;
+	const uint32_t newSessionDataLengthWithIoLimits = 35;
```
All of these are magic numbers. Consider to calculate them, with constexpr if possible to ensure compile time.
If they are not meant to be changed and the value is known at compile time, please use constexpr instead of const uint32_t.

rolysr replied:
Done.

lgsilva3087 commented on src/mount/mastercomm.cc:
```diff
@@ -476,261 +488,465 @@ void fs_register_config() {
+			fprintf(stderr, "got incorrect answer from sfsmaster\n");
 		} else {
-			safs_pretty_syslog(LOG_WARNING,"got incorrect answer from sfsmaster");
+			safs::log_warn("got incorrect answer from sfsmaster");
```
Looks more an error than a warning.
Consider if safs::log_err is more appropiate.
Applies to the others were return -1 is used.

rolysr replied:
Done. My idea is to leave it with older behavior for avoiding double printing, and then on a different PR addressed  a consistent modern way to print on the client using latest logger functions.

lgsilva3087 commented on src/mount/mastercomm.cc:
```diff
@@ -741,161 +957,91 @@ int fs_connect(bool verbose) {
+	const size_t saunafsVersionMajorLength = 2U;
+	const size_t saunafsVersionMinorLength = 1U;
+	const size_t saunafsVersionMicroLength = 1U;
+	const uint32_t fuseRegisterSessionTypeLength = 1U;
```
Compile time.

rolysr replied:
Done.

lgsilva3087 commented on src/mount/mastercomm.cc:
```diff
@@ -741,161 +957,91 @@ int fs_connect(bool verbose) {
+	uint32_t mapallgid = 0;
+	uint8_t mingoal = 0;
+	uint8_t maxgoal = 0;
+	uint32_t mintrashtime, maxtrashtime;
```
Please declare them in two lines and initialize them (this functions is not called frequently).

rolysr replied:
Done.

lgsilva3087 reviewed (commented):
Great improvement. Please see my suggestions.

lgsilva3087 commented on src/mount/mastercomm.cc:
```diff
@@ -476,261 +488,465 @@ void fs_register_config() {
+	       (messageValueIterator == newSessionDataLengthWithPackageVersion ||
+	        messageValueIterator == newSessionDataLengthWithMapAllUidGid ||
+	        messageValueIterator == newSessionDataLengthGoalsAndTrashtimes ||
+	        messageValueIterator == newSessionDataLengthWithIoLimits)))) {
```
Consider to use the DeMorgan's theorem here.
```cpp
if (messageValueIterator != 1 &&
	      (!gInitParams.meta ||
	       (messageValueIterator != newMetaSessionDataLengthLegacy &&
	        messageValueIterator != newMetaSessionDataLengthWithPackageVersion &&
	        messageValueIterator != newMetaSessionDataLengthWithGoalsAndTrashtimes)) &&
	      (gInitParams.meta != 0 ||
	       (messageValueIterator != newSessionDataLengthWithPackageVersion &&
	        messageValueIterator != newSessionDataLengthWithMapAllUidGid &&
	        messageValueIterator != newSessionDataLengthGoalsAndTrashtimes &&
	        messageValueIterator != newSessionDataLengthWithIoLimits))) {
```

rolysr replied:
Considered. But leaved as it is for keeping exact code idea. But they are equivalent, so the change is not required

lgsilva3087 reviewed (commented):
Please see my comment.

ralcolea commented on src/mount/mastercomm.cc:
```diff
@@ -476,261 +488,471 @@ void fs_register_config() {
+	const char *sessionFlagsNegativeStrTab[] = {SESFLAG_NEG_STRINGS};
+	bool clientHasSessionFlags = false;
+	std::stringstream infoToPrint;
+	for (uint32_t sesFlagsIndex = 0; sesFlagsIndex < sizeof(sesflags); sesFlagsIndex++) {
```
Consider separating the `for - block` with logic from the variables declaration.

rolysr replied:
Done.

ralcolea commented on src/mount/mastercomm.cc:
```diff
@@ -476,261 +488,471 @@ void fs_register_config() {
+	if (tcpnodelay(socket) < 0) {
 		if (verbose) {
-			fprintf(stderr,"can't connect to sfsmaster (\"%s\":\"%" PRIu16 "\")\n",masterstrip,masterport);
+			fprintf(stderr, "can't set TCP_NODELAY\n");
```
Consider using the C++ 23 more modern `std::println()` function and the new logging format for all the modified lines.

```suggestion
			std::println(stderr, "can't set TCP_NODELAY");
```

rolysr replied:
Considered. Maybe on another PR when we set a convention for this exact printing substitution.

ralcolea commented on src/mount/mastercomm.cc:
```diff
@@ -476,261 +488,471 @@ void fs_register_config() {
+			fprintf(stderr, "can't set TCP_NODELAY\n");
 		} else {
-			safs_pretty_syslog(LOG_WARNING,"can't connect to sfsmaster (\"%s\":\"%" PRIu16 "\")",masterstrip,masterport);
+			safs_pretty_syslog(LOG_WARNING, "can't set TCP_NODELAY");
```
Use `safs::log_warn()` instead.

rolysr replied:
Done.

ralcolea commented on src/mount/mastercomm.cc:
```diff
@@ -476,261 +488,469 @@ void fs_register_config() {
+	if (tcptoread(fd, registrationMessageBuffer.data(),
+	              kCltomaFuseRegisterHeaderLength + kFuseRegisterBlobAclPacketSizeValueLength,
+	              kDefaultTcpCommTimeoutMSeconds) !=
+	    (int32_t)(kCltomaFuseRegisterHeaderLength + kFuseRegisterBlobAclPacketSizeValueLength)) {
```
Consider caching the sum in a dedicated variable to avoid duplicating the sum and improve the readability.

rolysr replied:
Done.

lgsilva3087 reviewed (approved):
Looks ready. Please don't forget the changes intended for next PRs.

ralcolea commented on src/mount/mastercomm.cc:
```diff
@@ -476,261 +488,469 @@ void fs_register_config() {
+	get32bit(&messageFromMaster, messageValueIterator);
+	if (!(messageValueIterator == 1 ||
+	      (gInitParams.meta &&
+	       (messageValueIterator == kNewMetaSessionDataLengthLegacy ||
```
Consider simplifying this big if with less conditions by grouping groups of conditions in dedicated boolean expressions. Please, be careful when doing this, because with this big expressions it's very easy to make mistakes while transforming the original expression into a simpler one.

rolysr replied:
Done.

ralcolea commented on src/mount/mastercomm.cc:
```diff
@@ -741,161 +961,95 @@ int fs_connect(bool verbose) {
-		safs_pretty_syslog(LOG_NOTICE,"registered to master with new session (id #%" PRIu32 ")", sessionid);
+		safs_pretty_syslog(LOG_NOTICE, "registered to master with new session (id #%" PRIu32 ")",
+		                   sessionid);
 	}
```
Consider using the new format.

rolysr replied:
Done.

ralcolea reviewed (commented):
Very nice improvements in this version @rolysr!!! 👍 
Please, see some minor suggestions I shared for your consideration.

ralcolea commented on src/mount/mastercomm.cc:
```diff
@@ -476,261 +491,461 @@ void fs_register_config() {
+
+	get32bit(&messageFromMaster, messageValueIterator);
+	bool validNewMetaSessionDataLegth =
+	    kNewMetaSessionDataLengthLegacy ||
```
I think this line should be:

```suggestion
	    messageValueIterator == kNewMetaSessionDataLengthLegacy ||
```

rolysr replied:
Nice catch!

ralcolea reviewed (approved):
I think is ready. Great job @rolysr!!!
Please, confirm the last minor suggestion I have.
